Extension/src/LanguageServer/client.ts

```diff
@@ -1385,6 +1385,9 @@ export class DefaultClient implements Client {
         let filesEncodingChanged: boolean = false;
         if (workspaceFolder) {
             const lastFilesEncoding: PersistentFolderState<string> = new PersistentFolderState<string>("CPP.lastFilesEncoding", "", workspaceFolder);
+            if (lastFilesEncoding.Value === "") {
+                lastFilesEncoding.Value = filesEncoding;
```

Comment:
Could you just pass the default value in the PersistentFolderState constructor to the current filesEncoding value instead?